sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DaysWritable.scala

```diff
@@ -35,11 +35,12 @@ import org.apache.spark.sql.catalyst.util.DateTimeUtils.{rebaseGregorianToJulian
  * @param julianDays The number of days since the epoch 1970-01-01 in
  *                   Julian calendar.
  */
-private[hive] class DaysWritable(
+class DaysWritable(
     var gregorianDays: Int,
     var julianDays: Int)
   extends DateWritable {
```

Comment:
This causes a failure in `hive-1.2` profile due to the following.
- `org.apache.hadoop.hive.serde2.io.DateWritable`
- `org.apache.orc.storage.serde2.io.DateWritable`
```
/home/jenkins/workspace/SparkPullRequestBuilder@3/sql/core/v1.2/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcShimUtils.scala:54: type mismatch;
[error]  found   : org.apache.spark.sql.execution.datasources.DaysWritable
[error]  required: org.apache.orc.storage.serde2.io.DateWritable
[error]         result
[error]         ^
[info] org.apache.spark.sql.execution.datasources.DaysWritable <: org.apache.orc.storage.serde2.io.DateWritable?
[info] false
[error] /home/jenkins/workspace/SparkPullRequestBuilder@3/sql/core/v1.2/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcShimUtils.scala:57: type mismatch;
[error]  found   : org.apache.spark.sql.execution.datasources.DaysWritable
[error]  required: org.apache.orc.storage.serde2.io.DateWritable
[error]         new DaysWritable(getter.getInt(ordinal))
```

Reply:
We probably need to duplicate this class in v1.2 and v2.3 source code.

Reply:
I cloned `DaysWritable` for Hive ORC v1.2, and added a comment to the cloned class.